test/intest7.F90

```diff
@@ -0,0 +1,142 @@
+! check error messages, and using UFBPOS, UFBTAB, and VALX
+!
+! J. Ator, 2/23/2023
+
+module Share_errstr
+  ! This module is needed in order to share information between the test program and subroutine errwrt, because
+  ! the latter is not called by the former but rather is called directly from within the BUFRLIB software.
+
+  character*1500 errstr
+
+  integer errstr_len
+end module Share_errstr
+
+subroutine errwrt(str)
+  ! This subroutine supersedes the subroutine of the same name within the BUFRLIB software, so that we can
+  ! easily test the generation of error messages from within the library.
+
+  use Share_errstr
+
+  character*(*) str
+
+  integer str_len
+
+  str_len = len(str)
+  errstr ( errstr_len + 1 : errstr_len + str_len + 1 ) = str
+  errstr_len = errstr_len + str_len
+
+  return
+end subroutine errwrt
+
+program intest7
+  use Share_errstr
+
+  implicit none
+
+  integer*4 isetprm, igetprm, ireadns, ibfms
+
+  integer imgdt, iret1, iret2, jdate, nr8a, nr8v, nr8v2, idx1, idx2, nsub
+
+  integer mxr8pm, mxr8lv
+  parameter ( mxr8pm = 15 )
+  parameter ( mxr8lv = 5 )
+
+  real*8 r8arr (mxr8pm, mxr8lv), r8val
+
+  real valx
+
+  character cmgtag*8
+
+  print *, 'Testing reading IN_7 containing 2-03-YYY changed reference values, using inline ERRWRT'
+  print *, 'to check error messages, and using UFBPOS, UFBTAB, and VALX'
+
+#ifdef KIND_8
+  call setim8b ( .true. )
+#endif
+
+  ! Check error messages in ISETPRM.
+  iret1 = isetprm ( 'MXNRV', 5 )
+  errstr_len = 1
+  iret2 = isetprm ( 'DUMMY', 20 )
+  if ( ( iret1 .ne. 0 ) .or. ( iret2 .ne. -1 ) .or. &
+      ( index( errstr(1:errstr_len), 'ISETPRM - UNKNOWN INPUT PARAMETER DUMMY' ) .eq. 0 ) ) stop 1
+
+  ! Open the input file and DX table.
+  open ( unit = 11, file = 'testfiles/IN_7', form ='unformatted')
+  open ( unit = 12, file = 'testfiles/IN_7_bufrtab' )
+  call openbf ( 11, 'IN', 12 )
+  call openbf ( 11, 'QUIET', 1 )
+
+  ! Check error messages in IGETPRM.
+  iret1 = igetprm ( 'MXNRV' )
+  errstr_len = 1
+  iret2 = igetprm ( 'DUMMY' )
+  if ( ( iret1 .ne. 5 ) .or. ( iret2 .ne. -1 ) .or. &
+      ( index( errstr(1:errstr_len), 'IGETPRM - UNKNOWN INPUT PARAMETER DUMMY' ) .eq. 0 ) ) stop 2
+
+  ! Read some data values from the 1st messaage, which uses the 2-03-YYY operator to change one of the
+  ! reference values.
+  if ( ireadns ( 11, cmgtag, imgdt ) .ne. 0 ) stop 3
+  call ufbrep ( 11, r8arr, mxr8pm, mxr8lv, nr8a, 'TIDER' )
+  errstr_len = 1
+  call ufbrep ( 11, r8val, 1, 1, nr8v, 'DUMMY' )
+  idx1 = index( errstr(1:errstr_len), 'UFBREP - NO SPECIFIED VALUES READ IN' )
+  errstr_len = 1
+  call ufbrep ( 11, r8val, 0, 1, nr8v2, 'TIDER' )
+  idx2 = index( errstr(1:errstr_len), 'UFBREP - 3rd ARG. (INPUT) IS .LE. 0' )
+  if ( ( nr8a .ne. 2 ) .or. ( nr8v .ne. 0 ) .or. ( nr8v2 .ne. 0 ) .or. ( idx1 .eq. 0 ) .or. ( idx2 .eq. 0 ) &
+       .or. ( nint ( r8arr(1,1) ) .ne. -10000 ) .or. ( nint ( r8arr(1,2) ) .ne. 16 ) ) stop 4
```

Comment:
Some of these values like r8arr can be checked before later library calls. If they are incorrect, we want to exit immediately, not run additional library code.

Reply:
Again, good suggestion, and I'll make a quick pass through all of the intest codes to clean these up.  That will probably allow us to reduce the number of declared variables as well.

Reply:
Feel free to merge this PR and then do variable consolidation in another iteration...

Reply:
Thanks, but I'm going to have to push up another commit anyway to fix the key setting in developer.yml, so I may as well do all of this together ;-)